presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -86,14 +87,14 @@
     public HiveSplitManager(
             HiveConnectorId connectorId,
             HiveClientConfig hiveClientConfig,
-            ExtendedHiveMetastore metastore,
+            Function<HiveTransactionHandle, ExtendedHiveMetastore> metastoreProvider,
```

Comment:
I'm not sure why this works.  The provider method in `HiveClientModule` is declared as type `Function<HiveTransactionHandle, SemiTransactionalHiveMetastore>`.